cli/azd/cmd/env.go

```diff
@@ -276,6 +275,7 @@ type envNewAction struct {
 func newEnvNewAction(
 	azdCtx *azdcontext.AzdContext,
 	userProfileService *azcli.UserProfileService,
+	_ auth.LoggedInGuard,
```

Comment:
I am concern about this...  I think this is not a scalable approach..  Do we need to keep adding `Guards structs` as inputs for the constructor?
What if we need to return an error containing the result of all the `guards`?   

What about adding a `type AuthGuard func() error` and changing this line to be `authGuard AuthGuard`.

Then the first lines of `newEnvNewAction` like:
```
if err := authGuard(); err != nil {
   return err
}
```

then use `auth.LoggedInGuard` as the input here, to run the validation.  And register it to the IoC container as the lazy way to create `AuthGuard` .

Reply:
> What if we need to return an error containing the result of all the guards?

Let's deal with that if we need that flexibility. I can't imagine when we would not want `LoggedInGuard` to not return `not logged in`, and instead wanted `not logged in | and something else failed`. The tradeoff is more boilerplate vs. flexibility. Neither of those are tradeoffs that we can correctly value, since we don't know the exact requirements.